core/string/start_with_spec.rb

```diff
@@ -42,4 +42,34 @@
   it "works for multibyte strings" do
     "céréale".start_with?("cér").should be_true
   end
+
+  ruby_version_is "2.5" do
+    it "supports regexps" do
+      regexp = /[h1]/
+      "hello".start_with?(regexp).should be_true
+      "1337".start_with?(regexp).should be_true
+      "foxes are 1337".start_with?(regexp).should be_false
+    end
+
+    it "supports regexps with ^ and $ modifiers" do
+      regexp1 = /^\d{2}/
+      regexp2 = /\d{2}$/
+      "12test".start_with?(regexp1).should be_true
+      "test12".start_with?(regexp1).should be_false
```

Comment:
Do you think it would make sense to add another expectation for something along `"chunky\n12bacon"` (would return false in this case)

Reply:
I guess `"chunky\n12bacon".start_with?(/12/).should be_false` might be a good idea, just to be sure that it doesn't test for a line starting with, but the whole string starting with

Reply:
alright, I'll add that one